### PR TITLE
General: Update MacOs to PySide6

### PR DESCRIPTION
**Pull request by iLLiCiTiT**
## Brief description
Update PySide2 to PySide6 for MacOs and change all remaining imports of `Qt.py` module to `qtpy`. This change is needed for new MacOs distribution.

## Description
PR is combining change of Qt.py import with `qtpy` so all changes in this PR are affecting all hosts and it doesn't make sense to have it separated into multiple small PRs as most of them are connected to each other. Imports were changed in tools, utils widgets, launch script of non-python hosts, some vendor modules (`qargparse` and `scriptsmenu`) and global loader plugins. Added `QtPy` module to Python 2 vendor in latest available version that support `PySide` and `PyQt4`.

PySide6 changed few things here and there. E.g. how it is accessed to class methods (objects don't have access to class attributes) or renamed few methods. Some constants were removed and few classes were removed completelly.

We must make sure that all Qt tools in igniter are 100% right. Other possible bugs can be fixed later. Report any issue you will hit (make sure the issue was not already reported).

Checkstate constants are not integers anymore so output of `CheckStateRole` or `stateChanged` of checkbox can't be compared to `Checked`, `Unchecked` and `PartiallyChecked` without convertion to enum values first (big issue for backwards compatibility).

Issue with `qtpy` is that it always set `os.environ["QT_API"]` to best matching value. That causes issue if the value is set to `"pyside6"` because in that case older versions are not set. Currently was fixed for applications launch but it can cause more issues in future.

## TODOs
- try if PySide6 can be used on linux and MacOs

## Known issues
- [x] Loader does not show subsets -> Checkstate issue
- [ ] Settings UI stuck process -> Fix would be to "rewrite" settings, since the PR is affecting MacOs and Ayon switch will happen soon we're **ignoring this bug**
- [ ] Views have different background color of branch (probably stylesheets related) -> We'll have to revisit stylesheets in separated PR (there are reported other style issues)

## Testing notes:
All tools running in OpenPype process are potentially broken now and in most of cases the broken part is not recognized until someone hit exactly specific usecase. Because of `QtPy` module update and using it instead of `Qt.py` functionality in hosts should be checked too. The main focus is on tools running in OpenPype processes on MacOS.

It is required to install proper python and on MacOS also run `fetch_thirdpart_libraries` to install `PySide6`.

### Any possible UI tool executed from OpenPype tray
This consist of all tools from modules and in base of openpype.
- [x] LibraryLoader
- [x] Launcher
- [x] Local settings
- [x] Settings
- [x] ... others

### Loader actions
- [x] Copy filepath works
- [x] Copy file works
- [x] Delivery action works
- [x] Delete old version

### Hosts
In each host must be tested all of these (if the functionality is available there).
- Workfiles tool
    - shows work files
    - shows published files
    - Save as dialog works
    - Browse dialog works

- Loader tool
    - shows hierarchy, subsets, versions and representations
    - actions can be executed
    - actions with options will show dialog to confirm
    - links are shown

- Creator
    - creators are available
    - creation works

- Publisher
    - creators are available
    - attribute definitions are avialable
    - UI is updating as expected

- Publish Pype (pyblish-lite based)
    - plugins are shown
    - instances are shown
    - pages can be changed
    - details and terminal is available

- Subset manager
     - instances are visible
     - can be modified/removed

- Scriptsmenu
    - Menu is visible

### Hosts
- [ ] Maya
    - [x] 2020
    - [ ] 2022
    - [ ] 2023
- [ ] Nuke
    - [ ] 12.*
    - [x] 13.*
    - [x] 14.*
- [ ] Hiero
    - [ ] 12.*
    - [x] 13.*
    - [x] 14.*
- [ ] Houdini
    - [ ] with py 2
    - [ ] with py 3
    - [ ] Shelves can be added (not sure if it was possible)
- [x] Photoshop
- [x] AfterEffects
- [x] Harmony
- [ ] Resolve
- [ ] Fusion
- [ ] Unreal
- [ ] Blender
- [ ] TVPaint
- [ ] Flame
- [ ] Celaction
- [ ] 3ds Max

Write a comment if I forgot about some cases.

### CODE REVIEW

File: openpype/vendor/python/common/qargparse.py [hunk 1]
```diff
@@ -570,7 +570,7 @@ def data(self, index, role):
         model = QtCore.QStringListModel(self["default"])
         widget = QtWidgets.QListView()
         widget.setModel(model)
-        widget.setEditTriggers(widget.NoEditTriggers)
+        widget.setEditTriggers(QtWidgets.QAbstractItemView.NoEditTriggers)
```
Comment (BigRoy) [inline openpype/vendor/python/common/qargparse.py:573]: Why is this needed? According to the documentation QListView inherits from `QtWidgets.QAbstractItemView`: https://doc.qt.io/qtforpython/PySide6/QtWidgets/QAbstractItemView.html

And thus should also have the attribute `NoEditTriggers`?

Comment (iLLiCiTiT) [reply openpype/vendor/python/common/qargparse.py:573]: I will quote what in description "PySide6 changed few things here and there. E.g. how it is accessed to class methods (objects don't have access to class attributes)..." If you don't trust me you can try install PySide6 and try it on your own :)

**Comment by BigRoy** [comment]
For what it's worth - it seems that open-source compositor Natron is still using `PySide`? 

afbeelding

But there are open [Issues about its implementation of Qt in Natron 2.5.0](https://github.com/NatronGitHub/Natron/issues/854) and [long outstanding issue to update SDKs too](https://github.com/NatronGitHub/Natron/issues/681) so might be resolved with some time.

**Comment by iLLiCiTiT** [comment]
> For what it's worth - it seems that open-source compositor Natron is still using PySide?

PySide with Python 3.10.5? How? Why?

**Comment by BigRoy** [comment]
> PySide with Python 3.10.5? How? Why?

Not sure - [they are targeting Qt5, but with a next release cycle for 2.6+ it appears from this issue](https://github.com/NatronGitHub/Natron/issues/827). I personally have no idea how active Natron community is.

**Comment by kalisp** [comment]
In Nuke 13 `Library` doesn't fetch any subsets, even if `Load...` does.

**Comment by kalisp** [comment]
On Mac (M1) some tools are failing:
Creator with `qtpy.QtGui has no attribute 'QRegExpValidator'`
2023-01-13 18_12_17-516615434 - AnyDesk

SceneInventory with `SceneInventoryView has no attribute ExtendedSelection`
2023-01-13 18_12_59-516615434 - AnyDesk

Pyblish on `PluginType = QtGui.QStandardItem.UserType + 1`
2023-01-13 18_13_19-516615434 - AnyDesk

**Comment by 64qam** [comment]
Nuke 12 fails at startup with error 
image
```  File "/Applications/OpenPype 3.15.0.app/Contents/MacOS/openpype/vendor/python/python_2/qtpy/__init__.py", line 120, in <module>
    assert API in (PYQT5_API + PYQT4_API + PYSIDE_API + PYSIDE2_API)
AssertionError```

File: openpype/lib/applications.py [hunk 2]
```diff
@@ -908,24 +908,25 @@ def __init__(self, application, executable, env_group=None, **data):
...
+        if src_env is None:
+            src_env = os.environ
 
-        # subprocess.Popen keyword arguments
-        self.kwargs = {
-            "env": {
-                key: str(value)
-                for key, value in env.items()
-            }
+        ignored_env = {"QT_API",}
```
[the remark this answers is not in the document]
Comment (iLLiCiTiT) [reply openpype/lib/applications.py:922]:
```suggestion
        ignored_env = {"QT_API", }
```

**Comment by 64qam** [comment]
In Harmony Manager it doesn't show version when the version is changed. 
```Traceback (most recent call last):
  File "/Applications/OpenPype 3.14.10.app/Contents/MacOS/openpype/tools/utils/delegates.py", line 57, in paint
    style.CE_ItemViewItem, option, painter, option.widget
AttributeError: 'PySide6.QtWidgets.QCommonStyle' object has no attribute 'CE_ItemViewItem'
Traceback (most recent call last):
  File "/Applications/OpenPype 3.14.10.app/Contents/MacOS/openpype/tools/utils/delegates.py", line 57, in paint
    style.CE_ItemViewItem, option, painter, option.widget
AttributeError: 'PySide6.QtWidgets.QCommonStyle' object has no attribute 'CE_ItemViewItem'
Traceback (most recent call last):
  File "/Applications/OpenPype 3.14.10.app/Contents/MacOS/openpype/tools/utils/delegates.py", line 57, in paint
    style.CE_ItemViewItem, option, painter, option.widget
AttributeError: 'PySide6.QtWidgets.QCommonStyle' object has no attribute 'CE_ItemViewItem'
Traceback (most recent call last):
  File "/Applications/OpenPype 3.14.10.app/Contents/MacOS/openpype/tools/utils/delegates.py", line 57, in paint
    style.CE_ItemViewItem, option, painter, option.widget
AttributeError: 'PySide6.QtWidgets.QCommonStyle' object has no attribute 'CE_ItemViewItem'
  - { WrappedCallbackItem }: [  Running callback: <function HostListener.websocket_handler.<locals>.<lambda> at 0x148399d30>  ] 
```

**Comment by 64qam** [comment]
In Scene Inventory, Switch Asset still doesn't work:
```Traceback (most recent call last):
  File "/Applications/OpenPype 3.14.10.app/Contents/MacOS/openpype/tools/sceneinventory/view.py", line 293, in <lambda>
    lambda: self._show_switch_dialog(items))
  File "/Applications/OpenPype 3.14.10.app/Contents/MacOS/openpype/tools/sceneinventory/view.py", line 752, in _show_switch_dialog
    dialog = SwitchAssetDialog(self, items)
  File "/Applications/OpenPype 3.14.10.app/Contents/MacOS/openpype/tools/sceneinventory/switch_dialog.py", line 63, in __init__
    assets_combox = SearchComboBox(self)
  File "/Applications/OpenPype 3.14.10.app/Contents/MacOS/openpype/tools/sceneinventory/widgets.py", line 45, in __init__
    self.setInsertPolicy(self.NoInsert)
AttributeError: 'SearchComboBox' object has no attribute 'NoInsert'
  - { WrappedCallbackItem }: [  Running callback: <function HostListener.websocket_handler.<locals>.<lambda> at 0x12e9139d0>  ] ```

**Comment by 64qam** [comment]
In Harmony Scene Inventory there is also an issue with `Update to latest` template which errors:
```Traceback (most recent call last):
  File "/Applications/OpenPype 3.14.10.app/Contents/MacOS/openpype/tools/sceneinventory/view.py", line 246, in <lambda>
    lambda: _on_update_to_latest(items)
  File "/Applications/OpenPype 3.14.10.app/Contents/MacOS/openpype/tools/sceneinventory/view.py", line 230, in _on_update_to_latest
    update_container(item, -1)
  File "/Applications/OpenPype 3.14.10.app/Contents/MacOS/openpype/pipeline/load/utils.py", line 477, in update_container
    return loader.update(container, new_representation)
  File "/Applications/OpenPype 3.14.10.app/Contents/MacOS/openpype/pipeline/load/plugins.py", line 112, in update
    raise NotImplementedError("Loader.update() must be "
NotImplementedError: Loader.update() must be implemented by subclass
  - { WrappedCallbackItem }: [  Running callback: <function HostListener.websocket_handler.<locals>.<lambda> at 0x12e9139d0>  ] ```

**Comment by iLLiCiTiT** [comment]
> In Harmony Scene Inventory there is also an issue with Update to latest template which errors:

This is issue of the harmony loader (I can't do much about it).

**Review by 64qam — APPROVED** [review]
